datafusion/expr/src/expr.rs

```diff
@@ -1370,13 +1142,21 @@ pub struct Placeholder {
     /// The identifier of the parameter, including the leading `$` (e.g, `"$1"` or `"$foo"`)
     pub id: String,
     /// The type the parameter will be filled in with
-    pub data_type: Option<DataType>,
+    pub field: Option<FieldRef>,
 }
 
 impl Placeholder {
     /// Create a new Placeholder expression
     pub fn new(id: String, data_type: Option<DataType>) -> Self {
```

Comment:
I think we should probably deprecate this one too perhaps -- and direct users to the one below